File: source/zombie/text/templating/ITemplateBuilder.java
// Decompiled on Sat Jan 17 08:24:00 EST 2026 with Zomboid Decompiler v0.2.3 using Vineflower.
package zombie.text.templating;

import se.krka.kahlua.j2se.KahluaTableImpl;

/**
 * TurboTuTone.
 */
public interface ITemplateBuilder {
    String Build(String input);

    String Build(String input, IReplaceProvider replaceProvider);

    String Build(String input, KahluaTableImpl table);

    void RegisterKey(String key, KahluaTableImpl table);

    void RegisterKey(String key, IReplace replace);

    void Reset();
}

File: converted/src/zombie/text/templating/ITemplateBuilder.cpp
#include "zombie/text/templating/ITemplateBuilder.h"

namespace zombie {
namespace text {
namespace templating {

std::string ITemplateBuilder::Build(const std::string& input) {
 // TODO: Implement Build
 return "";
}

std::string ITemplateBuilder::Build(const std::string& input, IReplaceProvider replaceProvider) {
 // TODO: Implement Build
 return "";
}

std::string ITemplateBuilder::Build(const std::string& input, KahluaTableImpl table) {
 // TODO: Implement Build
 return "";
}

void ITemplateBuilder::RegisterKey(const std::string& key, KahluaTableImpl table) {
 // TODO: Implement RegisterKey
}

void ITemplateBuilder::RegisterKey(const std::string& key, IReplace replace) {
 // TODO: Implement RegisterKey
}

void ITemplateBuilder::Reset() {
 // TODO: Implement Reset
}

} // namespace templating
} // namespace text
} // namespace zombie
